### ISSUE TITLE
Add register endpoint

### ISSUE THREAD

**Issue by donni106** [ikuseiGmbH/smart-village-app-humhub-ext#2]
Currently it is not possible to register per API. Like `/auth/login` we need a new POST endpoint `/auth/register` that can receive at least `username`, `password`, `password`, `passwordconfirm` and optional `firstname`, `lastname`. New users should be added to the HumHub "Default Group" automatically.

It should return the same as the login request, so that users are logged in directly after registration:

```
{
  "code": 200,
  "message": "Success",
  "auth_token": "eyJ0e...tUx3Q",
  "expired_at": 0
}
```

**Comment by handysolver** [comment]
Hi there,
We notice that the registration option is already there if using the Humhub REST API 
More at: https://www.humhub.com/en/marketplace/rest/docs/html/user.html#tag/User/paths/~1user/post

Is there something more specific or a different type of registration you require?

<img width="1433" alt="Screenshot 2022-04-19 at 3 27 29 PM" src="https://user-images.githubusercontent.com/4486739/163979705-9f16054c-6dc3-4a3e-8392-95bf63f065a7.png">

**Comment by gaurav-handysolver** [comment]
Hi There,
also there is an api where we can add members to a particular group by specifying user id and group id.

**Comment by donni106** [comment]
@gaurav-handysolver hi! thanks for checking out. The `/user` endpoint does not work, as it requires an authentication. This endpoint can be used if some kind of admins want to add users, but not for first registrations.

without auth it results in:
```
{
  "name": "Unauthorized",
  "message": "Invalid token!",
  "code": 0,
  "status": "401",
  "type": "yii\\web\\HttpException"
}
```

**Comment by gaurav-handysolver** [comment]
Hi @donni106,
Registration API of user without auth is now created. we create a module called "smart village humhub api" and also an api in doc/postman folder. you can import that json file and test it.   
We also prepare a doc for setup the module. so, you can take help from that doc also.
Thanks!
 https://docs.google.com/document/d/1951Kw6rtRjlC_o2tVex2Tf7H1DXThGE7IOqE6lkLGes/edit#heading=h.ta8fiqb2277d

**Comment by gaurav-handysolver** [comment]
Hi there,
We made the registration endpoint api. from which user can register  without any auth bearer token by entering some basic details like username,email, password. firstname and lastname are optional as you mentioned eariler.

After user registered their account, he will be automatically added in default group of humhub which is "USER" group whose has (ID=2).

And Finally, username and password of the user will be authenticate by login function and if the username and password will be correct then the success message will be shown, otherewise it will thown an error message.

We also attached a screenshot where we are trying to register the user without token.
![image](https://user-images.githubusercontent.com/103625795/164445349-346d2df3-6691-4592-97d5-9b088d8e81f0.png)
